src/main/java/org/kainos/ea/api/ResponsibilityService.java

```diff
@@ -9,16 +9,19 @@
 import java.util.List;
 
 public class ResponsibilityService {
+    private ResponsibilityDao responsibilityDao;
 
-    private ResponsibilityDao responsibilityDao = new ResponsibilityDao();
+    public ResponsibilityService(ResponsibilityDao responsibilityDao) {
+        this.responsibilityDao = responsibilityDao;
+    }
 
     public List<Responsibility> getRoleResponsibilityById(int id) throws FailedToGetRoleResponsibilityException,
             RoleResponsibilityDoesNotExistException {
         List<Responsibility> responsibilitiesList = null;
         try {
             responsibilitiesList = responsibilityDao.getRoleResponsibilityById(id);
 
-            if (responsibilitiesList.size() == 0) {
+            if (responsibilitiesList == null) {
```

Comment:
Since getRoleResponsibilityById returns a list the check for size was correct, you never return null here only a list with size 0 or greater.

You could use` responsibilitiesList.isEmpty()` to make this clearer

Reply:
responsibilitiesList.isEmpty() and .size() both cause a null pointer exception, causing the test to fail, this was the only way i was able to get the tests to run without this error happening (although if you have any ideas as to how this can be fixed i would really appreciate it)